src/Build/Microsoft.Build.csproj

```
@@ -681,22 +681,24 @@
     <NuGetFrameworkWrapperRedirects_FilePath>$(IntermediateOutputPath)NuGetFrameworkWrapper.redirects.cs</NuGetFrameworkWrapperRedirects_FilePath>
   </PropertyGroup>
 
-  <!-- Extract binding redirects for Microsoft.Build from MSBuild.exe.config into a source file -->
+  <!-- Extract binding redirects for MSBuild and dependencies from MSBuild.exe.config into a source file.
+       This allows us to create secondary AppDomains with the same redirects at run-time, see
+       https://github.com/dotnet/msbuild/blob/main/documentation/NETFramework-NGEN.md#nugetframeworks -->
   <Target Name="GenerateAppDomainConfig"
           Inputs="..\MSBuild\app.config;..\MSBuild\app.amd64.config"
```

Built this locally and noticed an incremental build bug which would be fixed by

```suggestion
          Inputs="..\MSBuild\app.config;..\MSBuild\app.amd64.config;$(MSBuildThisFileFullPath)"
```

(to pick up modified logic in this target--unfortunately we can't be more granular than that currently)

---

Applied, thank you!